### PR TITLE
lib/gstreamer: use exe2os and filepath2os

### PR DESCRIPTION
**Pull request by tong1wu**
Signed-off-by: Tong Wu <tong1.wu@intel.com>

### CODE REVIEW

**Comment by uartie** [comment]
@tong1wu please fix test regressions.  HINT: don't use f-strings in `gen_xxx` functions.  The `gen_xxx` functions usually only return a "templated" string to be formatted later by caller.  That is, caller may use different values with the same returned template (e.g. r2r).

**Comment by tong1wu** [comment]
> @tong1wu please fix test regressions. HINT: don't use f-strings in `gen_xxx` functions. The `gen_xxx` functions usually only return a "templated" string to be formatted later by caller. That is, caller may use different values with the same returned template (e.g. r2r).

Thanks for pointing out. I have replaced the f-strings with a new variable. Please help review again. Thanks.

**Review by uartie — CHANGES_REQUESTED** [review]
Still alot of test failures.  Please don't use `self.decoded`, etc. in `gen_xxx` functions at all.  It might help to look at the ffmpeg classes to see how this is achieved for windows support.

File: lib/gstreamer/vppbase.py [hunk 1]
```diff
@@ -50,6 +51,7 @@ def gen_input_opts(self):
     return opts
 
   def gen_output_opts(self):
+    self.osdecoded = filepath2os(self.decoded)
```
Comment (uartie) [inline lib/gstreamer/vppbase.py:54]: This should only be done in the `vpp()` function each time after `self.decoded` is assigned.

Comment (uartie) [reply lib/gstreamer/vppbase.py:54]: I know the current design can be confusing to know when and where certain variables should be accessed.  I'm hoping to clean this confusion up in some upcoming refactors soon.  In the meantime, we just have to be careful.

**Comment by uartie** [comment]
I've changed this to "draft" since it's clear it has not been properly verified locally nor any help requested.

Once you're confident it works for you locally, you can switch it back to "ready for review".  If you need any help, I'd be happy to answer your questions... just ask.

**Comment by tong1wu** [comment]
Tested all components. The can work now. Ready for review thanks.

**Review by uartie — APPROVED** [review]

**Pull request merged**